### PR TITLE
Bug Fix[DB-12004]: SSL support for Oracle Assess Migration command

### PR DESCRIPTION
**Pull request by sanyamsinghal**
- tnsAdmin flag needs to be enabled for oracle assess-migration
- now it is passed to gather script for queries and ora2pg schema gather
- previously help messages were written wrt PG support only, modified them to refer both (PG and Oracle)

Testing:
1. Manually tested oracle ssl tests from our automation
2. Ran all oracle related ssl tests through automation
3. Manually tested new flags `--oracle-db-sid` and `--oracle-tns-alias`

### CODE REVIEW

**Comment by sanyamsinghal** [comment]
https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/2884/

File: yb-voyager/cmd/assessMigrationCommand.go [hunk 1]
```diff
@@ -520,7 +529,12 @@ func gatherAssessmentMetadataFromOracle() (err error) {
 	}
 
 	log.Infof("using script: %s", scriptPath)
-	return runGatherAssessmentMetadataScript(scriptPath, []string{"ORACLE_PASSWORD=" + source.Password},
+	tnsAdmin, err := getTNSAdmin(source)
+	if err != nil {
+		return fmt.Errorf("error getting tnsAdmin: %v", err)
+	}
+	log.Infof("tnsAdmin passed to oracle gather metadata script: %s", tnsAdmin)
+	return runGatherAssessmentMetadataScript(scriptPath, []string{"ORACLE_PASSWORD=" + source.Password, "TNS_ADMIN=" + tnsAdmin},
```
Comment (makalaaneesh) [inline yb-voyager/cmd/assessMigrationCommand.go:537]: I would suggest:
1. `runGatherAssessmentMetadataScript` should start with `env = os.Environ()` and then append whatever you want to append. We do that with all other commands that we run.
2. For TNS_ADMIN env var, makes sense what you're doing. Let's explicitly retrieve it and pass it. 
2. We should do the same for ORACLE_HOME. The priority should be oracleHome voyager flag, env var ORACLE_HOME, default `/usr/lib/oracle/21/client64`. 

(Something for the future: I don't see why we need to have oracle-home as a flag. We should just pick up the env var `ORACLE_HOME`)

File: yb-voyager/src/srcdb/data/gather-assessment-metadata/oracle/yb-voyager-oracle-gather-assessment-metadata.sh [hunk 2]
```diff
@@ -97,43 +97,50 @@ parse_oracle_dsn() {
...
-                } else if ($i ~ /SID *=/) {
-                    split($i, a, "=");
-                    sid=a[2];
-                    gsub(/^[ \t]+|[ \t]+$/, "", sid);
-                } else if ($i ~ /PORT *=/) {
-                    split($i, a, "=");
-                    port=a[2];
-                    gsub(/^[ \t]+|[ \t]+$/, "", port);
+    # Check if the connection string uses TNS alias format
+    if [[ $conn_str == *@* ]] && [[ $conn_str != *"(DESCRIPTION="* ]]; then
```
[the remark this answers is not in the document]
Comment (makalaaneesh) [reply yb-voyager/src/srcdb/data/gather-assessment-metadata/oracle/yb-voyager-oracle-gather-assessment-metadata.sh:101]: Probably still not fool-proof, but how about :
- figure out the part in the connection string after `@`
- if there is NO `/` or `DESCRIPTION=` in the connection string, assume TNS_ALIAS. 

Essentially, just add the condition for `/` for now.

[the remark this answers is not in the document]
Comment (sanyamsinghal) [reply yb-voyager/src/srcdb/data/gather-assessment-metadata/oracle/yb-voyager-oracle-gather-assessment-metadata.sh:101]: Yeah, `user/password@hostname:port/service_name` can also be one way to pass connection params

But we need to parse this uri format also to pass it to ORACLE_DSN in ora2pg conf file.
So i thought of supporting only two formats in our scripts(also mentioned in help message):
```
'username@(DESCRIPTION = (ADDRESS = (PROTOCOL = TCP)(HOST = hostname)(PORT = port))(CONNECT_DATA = (SID = SID)))'
or 'username@tns_alias'
```

[the remark this answers is not in the document]
Comment (sanyamsinghal) [reply yb-voyager/src/srcdb/data/gather-assessment-metadata/oracle/yb-voyager-oracle-gather-assessment-metadata.sh:101]: what do you think @makalaaneesh ? should we support more syntaxes of conn_uri here

File: yb-voyager/cmd/assessMigrationCommand.go (hunk 1 above)
Comment (sanyamsinghal) [reply yb-voyager/cmd/assessMigrationCommand.go:537]: ok, i believe after that we won't need to pass the `PATH` variable separately.

Comment (sanyamsinghal) [reply yb-voyager/cmd/assessMigrationCommand.go:537]: doing the change

File: yb-voyager/cmd/assessMigrationCommand.go [hunk 3]
```diff
@@ -570,8 +585,8 @@ func findGatherMetadataScriptPath(dbType string) (string, error) {
 func runGatherAssessmentMetadataScript(scriptPath string, envVars []string, scriptArgs ...string) error {
 	cmd := exec.Command(scriptPath, scriptArgs...)
 	log.Infof("running script: %s", cmd.String())
+	cmd.Env = os.Environ()
 	cmd.Env = append(cmd.Env, envVars...)
```
[the remark this answers is not in the document]
Comment (sanyamsinghal) [reply yb-voyager/cmd/assessMigrationCommand.go:589]: VERIFIED

File: yb-voyager/src/srcdb/data/gather-assessment-metadata/oracle/yb-voyager-oracle-gather-assessment-metadata.sh [hunk 4]
```diff
@@ -97,43 +98,50 @@ parse_oracle_dsn() {
...
+                for (i = 1; i <= NF; i++) {
+                    if ($i ~ /HOST *=/) {
+                        split($i, a, "=");
+                        host=a[2];
+                        gsub(/^[ \t]+|[ \t]+$/, "", host);
+                    } else if ($i ~ /SERVICE_NAME *=/) {
+                        split($i, a, "=");
+                        service_name=a[2];
+                        gsub(/^[ \t]+|[ \t]+$/, "", service_name);
+                    } else if ($i ~ /SID *=/) {
```
[the remark this answers is not in the document]
Comment (sanyamsinghal) [reply yb-voyager/src/srcdb/data/gather-assessment-metadata/oracle/yb-voyager-oracle-gather-assessment-metadata.sh:119]: tested by setting `--oracle-db-sid` flag and providing empty string for `--source-db-name`

**Review by makalaaneesh — APPROVED** [review]
LGTM. pls ensure oracle ssl tests are passing

**Pull request merged**